src/nnet3/nnet-cudnn-simple-component.cc

```diff
@@ -27,14 +28,61 @@
 
 namespace kaldi {
 namespace nnet3 {
-void array_strides(int n_dims, const int *shape, int *strides) {
-  int stride = 1;
-  for (int i = n_dims-1; i >= 0; --i) {
-    strides[i] = stride;
-    stride *= shape[i];
-  }
+
+/* The following functions are used to fill up the stride array
+   according to different tensor format. The order of the dimensions
+   in the shape array is by default NCXYZ, but the corresponding
+   position in the stride array determines the layout in the tensor.
+*/
+
+void array_strides_NXYZC(int n_dims, const int *shape, int *strides) {
+  strides[0] = shape[1] * shape[2] * shape[3] * shape[4];
+  strides[1] = 1;
+  strides[2] = shape[1] * shape[3] * shape[4];
+  strides[3] = shape[1] * shape[4];
+  strides[4] = shape[1];
+}
+
+void array_strides_NXZYC(int n_dims, const int *shape, int *strides) {
```

Comment:
Sorry that i forget to remove this, but this will be useful if we allow user to select the output tensor layout.
Anyway, as you have suggested, these array_strides_xxxx() should be moved to another file if we are adding more cudnn routine (e.g. pooling).